lib/valueflow.cpp

```diff
@@ -2584,10 +2583,15 @@ static void valueFlowAfterCondition(TokenList *tokenlist, SymbolDatabase* symbol
                                 bailout(tokenlist, errorLogger, after, "possible noreturn scope");
                             continue;
                         }
-                        if (check_if || !isReturnScope(after))
-                            values = &true_values;
+                        dead_else = isReturnScope(after);
                     }
 
+                    std::list<ValueFlow::Value> * values = nullptr;
```

Comment:
do we really want to only pick 1 values either from if or else? If both check_if and check_else are true I envision we can append the values.

Reply:
but well.. that is fixing FN and it is better that is done separately.

Reply:
I can look into that, but I would like to fix the regression first.